local-volume/provisioner/pkg/controller/controller.go

```diff
@@ -17,36 +17,41 @@ limitations under the License.
 package controller
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/golang/glog"
 
 	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/cache"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/common"
 	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/deleter"
 	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/discovery"
 	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/populator"
-	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/types"
 	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/util"
 
 	"k8s.io/client-go/kubernetes"
 )
 
-func StartLocalController(client *kubernetes.Clientset, config *types.UserConfig) {
+func StartLocalController(client *kubernetes.Clientset, config *common.UserConfig) {
 	glog.Info("Initializing volume cache\n")
 
-	runtimeConfig := &types.RuntimeConfig{
+	runtimeConfig := &common.RuntimeConfig{
 		UserConfig: config,
 		Cache:      cache.NewVolumeCache(),
 		VolUtil:    util.NewVolumeUtil(),
 		APIUtil:    util.NewAPIUtil(client),
 		Client:     client,
-		// TODO: make this unique based on node name?
-		Name: "local-volume-provisioner",
+		Name:       fmt.Sprintf("local-volume-provisioner-%v", config.Node.UID),
```

Comment:
I believe it should check if node name is the same when registering.  So the only case I can think of is if you delete the node, and then create a new node with the same name.  Here is one scenario:

1. On old node, have PV-a using path1
2. Delete old node (without deleting PV-a)
3. Create new node with same name, create PV-b using path1 and start using it
4. Delete PV-a, now provisioner will delete contents from the new node's path1, which is PV-b.

I think it would be better to protect data for as many cases as we can, even if we cannot protect from all situations.  Maybe for debugging/troubleshooting, the provisioner name can include the human readable name and the uid?

I will also document this limitation in the alpha user doc.